components/calendar/src/types.rs

```diff
@@ -41,37 +51,99 @@ pub struct FormattableYear {
 
     /// The year number in the current era (usually 1-based).
     pub number: i32,
+}
 
-    /// The year in the current cycle for cyclic calendars (1-indexed)
-    /// can be set to `None` for non-cyclic calendars
-    ///
-    /// For chinese and dangi it will be
-    /// a number between 1 and 60, for hypothetical other calendars it may be something else.
-    pub cyclic: Option<NonZeroU8>,
+/// Representation of a year as needed for formatting
+#[derive(Copy, Clone, Debug, PartialEq)]
+#[non_exhaustive]
+pub struct FormattableYear {
+    /// The "extended year", typically anchored with year 1 as the year 1 of either the most modern or
+    /// otherwise some "major" era for the calendar
+    pub extended_year: i32,
+    /// The rest of the details about the year.
+    pub kind: FormattableYearKind,
+}
 
-    /// The related ISO year. This is normally the ISO (proleptic Gregorian) year having the greatest
-    /// overlap with the calendar year. It is used in certain date formatting patterns.
+/// The type of year: Calendars like Chinese don't have an era and instead format with cyclic years.
+#[derive(Copy, Clone, Debug, PartialEq)]
+#[allow(clippy::exhaustive_enums)] // Only two kinds of years
+pub enum FormattableYearKind {
+    /// An Era and a numeric year in that era.
+    EraYear(Era, i32),
+    /// A cyclic year, and the related ISO year
     ///
-    /// Can be `None` if the calendar does not typically use `related_iso` (and CLDR does not contain patterns
-    /// using it)
-    pub related_iso: Option<i32>,
+    /// Knowing the cyclic year is typically not enough to pinpoint a date, however cyclic calendars
+    /// don't typically use eras, so disambiguation can be done by saying things like "Year 甲辰 (2024)"
+    Cyclic(NonZeroU8, i32),
```

Comment:
Suggestion: name the fields
```suggestion
    Cyclic {
        year_number: NonZeroU8,
        related_iso: i32,
    },
```